openssl-dynamic/src/main/c/jnilib.c

```diff
@@ -699,20 +269,32 @@ static void JNI_OnUnload_netty_tcnative0(JavaVM* vm, void* reserved) {
 
 // Invoked by the JVM when statically linked
 JNIEXPORT jint JNI_OnLoad_netty_tcnative(JavaVM* vm, void* reserved) {
-    return JNI_OnLoad_netty_tcnative0(vm, reserved);
+    tcn_global_vm = vm;
+    jint ret = netty_jni_util_JNI_OnLoad(vm, reserved, "netty_tcnative", netty_internal_tcnative_Library_JNI_OnLoad);
+    if (ret == JNI_ERR) {
+        tcn_global_vm = NULL;
```

Comment:
followup PR: we should kill this static state and pass the `JNIEnv*` provided by each JNI call. Removal of static state can simplify state lifetime and potential confusion (e.g. what if the `OnUnload` method is called and depends upon this).

Reply:
I will to check if this is possible as this is used without callbacks that are called by openssl... but generally I agree